admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -929,25 +929,58 @@ var form = (function() {
         });
       };
 
-      /** On event "tokenfield:createtoken" : stop event if its not a typehead result */
+      /** On event "tokenfield:createtoken" : check values are valid if its not a typehead result */
       $('#form_step3_attributes').on('tokenfield:createtoken', function(e) {
-        if (!e.attrs.data && e.handleObj.origType !== 'tokenfield:createtoken') {
-          return false;
+        if (!e.attrs.data)
+          if (e.handleObj.origType !== 'tokenfield:createtoken') {
+            return false;
+          }
+          var orgLabel = e.attrs.label;
+          if (e.attrs.label == e.attrs.value) {
+            engine.search(e.attrs.label, function(result) {
+              if (result.length >= 1) {
+                e.attrs.label = result[0].label;
+                e.attrs.value = result[0].value;
+                e.attrs.data = [];
+                e.attrs.data['id_group'] = result[0].data.id_group;
+              }
+            });
+          } else {
+            var attr = $(`.js-attribute-checkbox[data-value="${e.attrs.value}"]`);
+
+            if (attr) {
+              e.attrs.label = attr.data('label');
+              e.attrs.value = attr.data('value');
+              e.attrs.data = [];
+              e.attrs.data['id_group'] = attr.data('group-id');
+            }
+          }
+
+          if(e.attrs.data){
+            if(filter([e.attrs]).length == 0){
```

Comment:
```suggestion
            if(filter([e.attrs]).length === 0){
```

In that case that's unnecessary but as we want to keep consistance in our codebase, can you prefer using strict operators ?